spark/src/test/java/io/ddf/spark/SparkDDFManagerTests.java

```diff
@@ -46,4 +59,163 @@ public void testSimpleSparkDDFManager() throws DDFException {
+    LOG.info("========== testFolder/folder ==========");
+    S3DDF folderDDF = s3DDFManager.newDDF("jing-bucket", "testFolder/folder/", "year int, value int", null);
+    DDF folderSparkDDF = sparkDDFManager.copyFrom(folderDDF);
+    LOG.info(folderSparkDDF.sql("select * from @this", "error").getRows().toString());
+    assert(folderSparkDDF.getNumRows() == 4);
+
+    // Test copy from a folder with all json files.
+    LOG.info("========== testFolder/allJson ==========");
+    S3DDF allJsonDDF = s3DDFManager.newDDF("jing-bucket", "testFolder/allJson/", null, null);
+    DDF allJsonSparkDDF = sparkDDFManager.copyFrom(allJsonDDF);
+    // TODO: is it 6?
+    // assert(allJsonSparkDDF.getNumRows() == 4);
+
+    LOG.info("========== testFolder/folder/d.json ==========");
+    S3DDF jsonDDF = s3DDFManager.newDDF("adatao-test", "json/", null, null);
+    DDF jsonSparkDDF = sparkDDFManager.copyFrom(jsonDDF);
+    LOG.info(jsonSparkDDF.sql("select * from @this limit 5", "error").getRows().toString());
+
+
+    // Copy From a csv, the schema is not given.
+    LOG.info("========== testFolder/hasHeader.csv ==========");
+    Map<String, String> options = new HashMap<>() ;
+    options.put("header", "true");
+    S3DDF csvDDF = s3DDFManager.newDDF("jing-bucket", "testFolder/hasHeader.csv", null, options);
+    DDF csvSparkDDF = sparkDDFManager.copyFrom(csvDDF);
+    LOG.info(csvSparkDDF.sql("select * from @this", "error").getRows().toString());
+    assert(csvSparkDDF.getNumRows()==2);
+
+
+    // Copy From a csv, the schema is not given, and has no header.
+    LOG.info("========== testFolder/noHeader.csv ==========");
+    S3DDF csvDDF2 = s3DDFManager.newDDF("jing-bucket", "testFolder/noHeader.csv", null, null);
+    DDF csvSparkDDF2= sparkDDFManager.copyFrom(csvDDF2);
+    LOG.info(csvSparkDDF2.sql("select * from @this", "error").getRows().toString());
+    assert(csvSparkDDF2.getNumRows()==2);
+
+    // Copy From a csv, the schema is given and has no header.
+    LOG.info("========== testFolder/noHeader.csv ==========");
+    S3DDF csvDDF3 = s3DDFManager.newDDF("jing-bucket", "testFolder/noHeader.csv", "year int, val string", null);
+    DDF csvSparkDDF3 = sparkDDFManager.copyFrom(csvDDF3);
+    LOG.info(csvSparkDDF3.sql("select * from @this", "error").getRows().toString());
+    assert (csvSparkDDF3.getNumRows()==2);
+
+
+    LOG.info("========== tsv ==========");
+    S3DDF tsvDDF = s3DDFManager.newDDF("adatao-sample-data", "test/tsv/noheader/results.tsv", "ID int, FlagTsunami " +
+        "string, Year int, " +
+        "Month int, Day int, Hour int, Minute int, Second double, FocalDepth int, EqPrimary double, EqMagMw double, EqMagMs double, EqMagMb double, EqMagMl double, EqMagMfd double, EqMagUnk double, Intensity int, Country string, State string, LocationName string, Latitude double, Longitude double, RegionCode int, Death int, DeathDescription int, Injuries int, InjuriesDescription int", null);
+    // TODO: Check the file?
```

Comment:
You can try this using old session.load_from_s3 API in PyClient, I'm sure this file is ok